IdentityCore/src/cache/accessor/MSIDSharedTokenCache.m

```diff
@@ -105,45 +124,46 @@ - (BOOL)saveTokensWithRequestParams:(MSIDRequestParameters *)requestParams
         }
         
         // If it's an FRT, save it separately and update the clientId of the token item
-        MSIDToken *familyRefreshToken = [refreshToken copy];
+        MSIDRefreshToken *familyRefreshToken = [refreshToken copy];
         familyRefreshToken.clientId = [MSIDTokenCacheKey familyClientId:refreshToken.familyId];
         
         return [self saveRefreshTokenInAllCaches:familyRefreshToken
                                      withAccount:account
                                          context:context
                                            error:error];
     }
-    else
+    else if ([_primaryAccessor supportsTokenType:MSIDTokenTypeLegacyADFSToken])
     {
         MSIDAdfsToken *adfsToken = [[MSIDAdfsToken alloc] initWithTokenResponse:response
-                                                                        request:requestParams
-                                                                      tokenType:MSIDTokenTypeAdfsUserToken];
+                                                                        request:requestParams];
         
-        MSIDAccount *adfsAccount = [[MSIDAccount alloc] initWithUpn:@""
-                                                               utid:nil
-                                                                uid:nil];
+        account.legacyUserId = @"";
         
         // Save token for ADFS
-        return [_primaryAccessor saveAccessToken:adfsToken
-                                       account:adfsAccount
-                                 requestParams:requestParams
-                                       context:context
-                                         error:error];
+        [_primaryAccessor saveToken:adfsToken
+                            account:account
+                            context:context
+                              error:error];
     }
+    
+    return [_primaryAccessor saveAccount:account
```

Comment:
I didn't see accountType being set for variable `account`? Did I miss anything? :)

Reply:
fixed :)